src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -66,30 +75,113 @@ public async Task<SecretResponseModel> GetAsync([FromRoute] Guid id)
         {
             throw new NotFoundException();
         }
+
+        if (!await userHasReadAccessToProject(secret))
+        {
+            throw new NotFoundException();
+        }
+
```

Comment:
Need to add the `_currentContext.AccessSecretsManager(organizationId)` check to this endpoint